examples/Darcy/darcy.jl

```diff
@@ -0,0 +1,179 @@
+# # [Learning the Pearmibility field in a Darcy flow from noisy sparse observations] 
+
+# In this example we hope to illustrate function learning. One may wish to use function learning in cases where the underlying parameter of interest is actual a finite-dimensional approximation (e.g. spatial discretization) of some "true" function. Treating such an object directly will lead to increasingly high-dimensional learning problems as the spatial resolution is increased, resulting in poor computational scaling and increasingly ill-posed inverse problems. Treating the object as a discretized function from a function space, one can learn coefficients not in the standard basis, but instead in a basis of this function space, it is commonly the case that functions will have relatively low effective dimension, and will be depend only on the spatial discretization due to discretization error, that should vanish as resolution is increased. 
```

Comment:
small typo: "parameter of interest is actual a" -> "parameter of interest is actually a".
My interpretation of this paragraph is as follows:
In this example, we hope to illustrate function learning. While traditional inverse problems involve the calibration of parameters of a predefined model, function learning allows one to learn a finite-dimensional approximation (e.g. spatial discretization) of some “true” function/model. When treating the approximation as just random samples at evaluation points, increasing the spatial resolution leads to increasingly high dimensional learning problems, thus giving poor computational scaling and increasingly ill-posed inverse problems. When treating the approximation instead as a discretized function from a function space, one can learn coefficients instead of a basis in this function space. Since it is commonly the case that functions have relatively low effective dimension in this space, the dependence on the spatial discretization only arises in discretization error, which vanishes as resolution is increased.

Reply:
I did not quite insert this, interpretation, but I hope that the new version is a good intermediary. Feel free to iterate again

Reply:
Also note that this comment was left on darcy.jl example, not on the darcy.md docs!